cmd/bootstrap/cmd/partner_infos.go

```diff
@@ -0,0 +1,244 @@
+	partnerStakesFileName = "partner-stakes.json"
+	flowNodeAddrPart      = "nodes.onflow.org"
+	defaultPartnerStake   = "100"
+)
+
+// NodePubInfo basic representation of node-pub-info.json data
+type NodePubInfo struct {
+	Role          string `json:"Role"`
+	Address       string `json:"Address"`
+	NodeID        string `json:"NodeID"`
+	Stake         string `json:"Stake"`
+	NetworkPubKey string `json:"NetworkPubKey"`
+	StakingPubKey string `json:"StakingPubKey"`
+}
+
+// PartnerStakesInfo mapping of NodeID =>
+type PartnerStakesInfo map[string]string
+
+// populatePartnerInfos represents the `populate-partner-infos` command which will read the proposed node
+// table from the staking contract and for each identity in the proposed table generate a node-info-pub
+// json file. It will also generate the partner-stakes json file.
+var populatePartnerInfosCMD = &cobra.Command{
+	Use:   "populate-partner-infos",
+	Short: "Generates a node-info-pub-*.json for all proposed identities in staking contract and corresponding partner-stakes.json file.",
+	Run:   populatePartnerInfosRun,
+}
+
+func init() {
+	rootCmd.AddCommand(populatePartnerInfosCMD)
+
+	populatePartnerInfosCMD.Flags().StringVar(&flagOutputDir, "out", "", "the directory where the generated node-info-pub files will be written")
+	populatePartnerInfosCMD.Flags().StringVar(&flagANAddress, "access-address", "", "the address of the access node used for client connections")
+	populatePartnerInfosCMD.Flags().StringVar(&flagANNetworkKey, "access-network-key", "", "the network key of the access node used for client connections in hex string format")
+	populatePartnerInfosCMD.Flags().StringVar(&flagNetworkEnv, "network", "", "the network string, expecting one of ( mainnet | testnet | localnet )")
+
+	cmd.MarkFlagRequired(populatePartnerInfosCMD, "out")
+	cmd.MarkFlagRequired(populatePartnerInfosCMD, "access-address")
+}
+
+// populatePartnerInfosRun generate node-pub-info file for each node in the proposed table and the partner stakes file, and prints the
+// address and node ID of any flow nodes that were skipped.
+func populatePartnerInfosRun(_ *cobra.Command, _ []string) {
+	ctx := context.Background()
+
+	env, err := common.EnvFromNetwork(flagNetworkEnv)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get environment for network (%s)", flagNetworkEnv)
+	}
+
+	flowClient := getFlowClient()
+
+	proposedNodeIDS, err := executeGetProposedTableScript(ctx, env, flowClient)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get proposed table")
+	}
+
+	partnerStakes := make(PartnerStakesInfo)
+	skippedNodes := 0
+	numOfNodesByType := map[string]int{
+		flow.RoleCollection.String():   0,
+		flow.RoleConsensus.String():    0,
+		flow.RoleExecution.String():    0,
+		flow.RoleVerification.String(): 0,
+		flow.RoleAccess.String():       0,
+	}
+	totalNumOfPartnerNodes := 0
+
+	for _, id := range proposedNodeIDS.Values {
+		info, err := executeGetNodeInfoScript(ctx, env, flowClient, id)
```

Comment:
We can put the for loop in the Cadence script and save a linear number of network calls. Here's a script that will do that:
```
import FlowIDTableStaking from 0x8624b52f9ddcd04a

pub fun main(): [FlowIDTableStaking.NodeInfo] {
    let nodeIDs = FlowIDTableStaking.getProposedNodeIDs()

    var infos: [FlowIDTableStaking.NodeInfo] = []
    for nodeID in nodeIDs {
        let node = FlowIDTableStaking.NodeInfo(nodeID: nodeID)
        infos.append(node)
    }

    return infos
}
```

Reply:
https://github.com/onflow/flow-go/pull/1988/commits/c34743b2afef68f94ebeeae98cdca9695a565740

Reply:
I don't think that commit addresses the comment. We still have a linear number of network calls. 

It might be fine as-is depending on how long the script typically takes with Mainnet-sized identity table, but moving the loop into Cadence could save us minutes off each script execution.

Reply:
Meant to put this in separate commit https://github.com/onflow/flow-go/pull/1988/commits/cba109821575449a48fc29c352bf60649e5a3249